03-viz.Rmd

```diff
@@ -1,40 +1,528 @@
+
+- *Common file types:* 
+    - [JPEG](https://en.wikipedia.org/wiki/JPEG) (`.jpg`, `.jpeg`): lossy, usually used for photographs 
+    - [PNG](https://en.wikipedia.org/wiki/Portable_Network_Graphics) (`.png`): lossless, usually used for plots / line drawings
+    - [BMP](https://en.wikipedia.org/wiki/BMP_file_format) (`.bmp`): lossless, raw image data, no compression (rarely used)
+    - [TIFF](https://en.wikipedia.org/wiki/TIFF) (`.tif`, `.tiff`): typically lossless, no compression, used mostly in graphic arts, publishing
+- *Free Editing software:* [GIMP](https://www.gimp.org/), [JS Paint](https://jspaint.app)
+
+**Vector / Scalable Graphics** images are represented as a collection of mathematical objects (lines, surfaces, shapes, curves). When the computer displays the image, it redraws all of the elements using their mathematical formulas.
+
+- *Common file types:* 
+    - [SVG](https://en.wikipedia.org/wiki/Scalable_Vector_Graphics) (`.svg`): general-purpose use 
+    - [EPS](https://en.wikipedia.org/wiki/Encapsulated_PostScript) (`.eps`), general-purpose use (rarely used)
+- *Free Editing software:* [Inkscape](https://inkscape.org/), [Boxy SVG](https://boxy-svg.com/)
+
+Raster and vector images have opposing advantages and disadvantages. A raster image of a fixed width / height takes the same amount of space and time to load regardless of 
+what the image shows (caveat: the compression algorithms may shrink the image more or run faster for certain images). A vector image takes space and time to load 
+corresponding to how complex the image is, since the computer has to draw all the elements each time it is displayed. For example, if you have a scatter plot with 1 million
+points stored as an SVG file, it may take your computer some time to open the image. On the other hand, you can zoom into / scale up 
+vector graphics as much as you like without the image looking bad, while raster images eventually start to look "pixellated." 
+
+> **PDF files:** 
+> The portable document format [PDF](https://en.wikipedia.org/wiki/PDF) (`.pdf`) is commonly used to
+> store *both* raster and vector graphics formats. If you try to open a PDF and it's taking a long time
+> to load, it may be because there is a complicated vector graphics image that your computer is rendering. 
+
+
+
+
+Let's investigate how different image file formats behave with a scatter plot of the
+[Old Faithful dataset](https://www.stat.cmu.edu/~larry/all-of-statistics/=data/faithful.dat), which happens to be available in base R under the name `faithful`:
+
+```{r 03-plot-line, collapse=TRUE}
+library(tidyverse)
```

Comment:
This package should be loaded at the beginning of the chapter, not here.

Reply:
moved it up